lib/logstash/outputs/elasticsearch_java.rb

```diff
@@ -162,37 +165,24 @@ class LogStash::Outputs::ElasticSearch < LogStash::Outputs::Base
   #     `["127.0.0.1:9200","127.0.0.2:9200"]`
   # It is important to exclude http://www.elastic.co/guide/en/elasticsearch/reference/current/modules-node.html[dedicated master nodes] from the `host` list
   # to prevent LS from sending bulk requests to the master nodes.  So this parameter should only reference either data or client nodes.
-
-  config :host, :validate => :array
+  config :hosts, :validate => :array, :default => ["127.0.0.1"]
 
   # The port for Elasticsearch transport to use.
   #
   # If you do not set this, the following defaults are used:
-  # * `protocol => http` - port 9200
   # * `protocol => transport` - port 9300-9305
   # * `protocol => node` - port 9300-9305
-  config :port, :validate => :string
+  config :port, :validate => :string, :default => "9300-9305"
 
   # The name/address of the host to bind to for Elasticsearch clustering
-  config :bind_host, :validate => :string
+  # This MUST be set when the node protocol is used. The internal Elasticsearch node
+  # will bind to this ip. This ip MUST be reachable by all nodes in the Elasticsearch cluster
+  config :network_host, :validate => :string
 
-  # This is only valid for the 'node' protocol.
+  # This is only valid for the 'node' protocol. This sets the port to bind to on 'network_host'
   #
   # The port for the node to listen on.
-  config :bind_port, :validate => :number
-
-  # Run the Elasticsearch server embedded in this process.
-  # This option is useful if you want to run a single Logstash process that
-  # handles log processing and indexing; it saves you from needing to run
-  # a separate Elasticsearch process. An example use case is 
-  # proof-of-concept testing.
-  # WARNING: This is not recommended for production use!
-  config :embedded, :validate => :boolean, :default => false
-
-  # If you are running the embedded Elasticsearch server, you can set the http
-  # port it listens on here; it is not common to need this setting changed from
-  # default.
-  config :embedded_http_port, :validate => :string, :default => "9200-9300"
+  config :transport_tcp_port, :validate => :number
```

Comment:
to me this implies TransportClient (bah, overloaded terminology!). Can you tell me a bit about the rename from `bind_port` to `transport_tcp_port` ?